aqueduct/gateway/views/mcp.py

```diff
@@ -0,0 +1,801 @@
+    await _ensure_session_manager_started()
+
+    if not session_id:
+        log.warning(f"Session ID required for MCP server '{name}'")
+        return JsonResponse({"error": "Session ID required"}, status=400)
+
+    session = await session_manager.get_session(session_id)
+    if not session:
+        log.warning(f"Session {session_id} not found for MCP server '{name}'")
+        return JsonResponse({"error": "Session not found"}, status=404)
+
+    if session.terminated:
+        log.warning(f"Session {session_id} is terminated for MCP server '{name}'")
+        return JsonResponse({"error": "Session terminated"}, status=410)
+    if not session.is_ready():
+        log.warning(
+            f"Session {session_id} not ready for MCP server '{name}' (state: {session.state.value})"
+        )
+        return JsonResponse(
+            {"error": f"Session not ready (state: {session.state.value})"}, status=503
+        )
+
+    log.info(f"MCP GET {name} - SSE stream for existing session {session_id}")
+
+    # Build headers with proper CORS origin
+    headers = {"Cache-Control": "no-cache", "Connection": "keep-alive"}
+
+    # Set CORS header only if origin is allowed
+    origin = request.headers.get("origin")
+    allowed_origins = getattr(settings, "MCP_ALLOWED_ORIGINS", [])
+
+    if origin:
+        # Check exact match first
+        if origin in allowed_origins:
+            headers["Access-Control-Allow-Origin"] = origin
+        else:
+            # Check wildcard patterns
+            for allowed in allowed_origins:
+                if allowed.endswith(":*"):
+                    base_origin = allowed[:-2]
+                    if origin.startswith(base_origin + ":"):
+                        headers["Access-Control-Allow-Origin"] = origin
+                        break
+
+    return StreamingHttpResponse(
+        streaming_content=_mcp_sse_stream(session_id),
+        content_type="text/event-stream",
+        headers=headers,
+    )
+
+
+async def handle_post_request(
+    request: ASGIRequest,
+    name: str,
+    json_rpc_message: JSONRPCMessage | None,
+    session_id: str | None,
+    is_initialize: bool,
+) -> JsonResponse:
+    """Send message to MCP session."""
+    import httpx
+    from anyio import ClosedResourceError
+
+    await _ensure_session_manager_started()
+
+    if json_rpc_message is None:
+        return JsonResponse({"error": "Missing JSON-RPC message"}, status=400)
+
+    message_type = json_rpc_message.root.__class__.__name__
+    log.info(f"MCP POST {name} - Session: {session_id}, Method: {message_type}")
+
+    try:
+        if is_initialize and not session_id:
+            log.info(f"Initializing session for MCP server '{name}'")
+            mcp_config = get_mcp_config()
+            server_config = mcp_config[name]
+            url = server_config["url"]
+            session_id = await session_manager.create_session(url)
+            log.info(f"Session {session_id} initialized")
```

Comment:
Wouldn't this log entry make more sense after we make sure that `session_id` is not `None`?